apps/api/src/app/integrations/usecases/get-novu-integration/get-novu-integration.usecase.ts

```diff
@@ -0,0 +1,97 @@
+import { Injectable } from '@nestjs/common';
+import { IntegrationEntity, MessageRepository, IntegrationRepository, ExecutionDetailsRepository } from '@novu/dal';
+import { GetNovuIntegrationCommand } from './get-novu-integration.command';
+import { ChannelTypeEnum, EmailProviderIdEnum, ExecutionDetailsStatusEnum } from '@novu/shared';
+import { startOfMonth, endOfMonth } from 'date-fns';
+
+@Injectable()
+export class GetNovuIntegration {
+  constructor(
+    private messageRepository: MessageRepository,
+    private integrationRepository: IntegrationRepository,
+    private executionDetailsRepository: ExecutionDetailsRepository
+  ) {}
+
+  async execute(command: GetNovuIntegrationCommand): Promise<IntegrationEntity[]> {
+    const providerId = this.getProviderId(command.channelType);
+
+    if (providerId === undefined) {
+      return [];
+    }
+
+    const sentMessagesCount = await this.executionDetailsRepository.count({
+      status: ExecutionDetailsStatusEnum.SUCCESS,
+      _organizationId: command.organizationId,
+      channel: command.channelType,
+    });
+
+    if (sentMessagesCount > 0) {
+      return [];
+    }
```

Comment:
I'm not sure about this piece of logic here, why is this check needed? Won't it stop using this integration after the first attempt of using it? I'm not fully understand why this is needed here.

Reply:
I think we should discuss the rules again after have seen your comments...